src/sidebar/components/test/group-list-test.js

```diff
@@ -332,4 +332,89 @@ describe('groupList', function () {
     assert.calledWith($window.open, 'https://test.hypothes.is/groups/new',
       '_blank');
   });
+  describe('#showGroupMenu', () => {
+    it('should set showGroupMenu=false if third party service and only one group', function () {
```

Comment:
Given that this is no longer just a test for the `#showGroupMenu` controller method, the test description isn't really accurate any more.

I can see two ways we could address this:

1. Create two separate sets of tests, one that checks the result of the `showGroupMenu` method depending on the group inputs, and one that checks the rendering depending on the result of the `showGroupMenu` method. The second set of tests could simply stub `showGroupMenu` to return a fixed value instead of repeating the setup.

2. Continue to have each test checking both the `showGroupMenu` result and the rendered output, but change the name of the `describe` block and `it` description to reflect the level of the "API" which is being tested, in this case the user interface. Something like this:

    ```js
    describe('group menu visibility', () => {
       it('is hidden when using third party accounts if there is only one group', () => { ... })
   
       it('is visible when using third party accounts if there are multiple groups', () => { ... }}
    });
    ```

If the logic for `showGroupMenu` was complex and there were many permutations to test, I would definitely split up the tests, because you'd still only need two tests for the rendering but you might have many more for the controller method behaviour. In this case, it doesn't matter as much.